core/src/main/java/nu/yona/server/goals/service/GoalService.java

```diff
@@ -108,19 +114,15 @@ public GoalDTO updateGoal(UUID userID, UUID goalID, GoalDTO newGoalDTO, Optional
 		User userEntity = userService.getUserEntityByID(userID);
 		Goal existingGoal = getGoalEntity(userEntity, goalID);
 
-		if (newGoalDTO.getCreationTime().isPresent())
+		verifyGoalUpdate(existingGoal, newGoalDTO);
+		if (newGoalDTO.getCreationTime().isPresent() && !newGoalDTO.isGoalChanged(existingGoal))
```

Comment:
The Mobiquity app is not setting the creation time, so that's not an issue here. I'll add that check, also for our own safety when writing tests.